sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -447,7 +447,11 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
       // Create the table if the table didn't exist.
       if (!tableExists) {
         val schema = JdbcUtils.schemaString(df, url)
-        val sql = s"CREATE TABLE $table ($schema)"
+        // To allow certain options to append when create a new table, which can be
+        // table_options or partition_options.
+        // E.g., "CREATE TABLE t (name string) ENGINE=InnoDB DEFAULT CHARSET=utf8"
+        val createtblOptions = extraOptions.getOrElse("createTableOptions", "")
```

Comment:
@HyukjinKwon  For those database specific options can be merged as createTableOptions. Not recommend to enumerate them one by one. Just as @srowen suggested.

Thanks to @HyukjinKwon and @srowen. I will propose a draft, and back to both of you later. Thanks.